Shopify/shopify-cli
# [token-based-auth] Support passing the connection information through arguments

Fixes https://github.com/Shopify/shopify-cli-planning/issues/75

### WHY are these changes introduced?
The `shopify extension push` command reads the connection information, which is necessary for deploying a script, from the `.env` file. Because that file contains sensitive data, some developers might not have it in their repository, and in that scenario deployments from CI/CD environments will fail because the file is missing.

### WHAT is this pull request doing?
Until we can revisit the `.env` block-based configuration in the new cohesive app development experience, I'm building a solution upon the existing `EnvFile` model that the in-memory project representation depends on. I'm parsing `api_key`, `api_secret`, and `registration_id` from the list of arguments passed to the command, and using them to override any value loaded from an existing `.env` file.

### VISION
The above is a solution that strikes a balance between addressing the need that we have vs not having to refactor the project and configuration loading logic. Long-term I think we should revisit this approach to minimize side-effects and assumptions on the project being in the current working directory. In my experience, commands can be easily modeled as a set of functions where the first one is the loading of the project and its configuration. The project is then passed through a pipeline of operators that do something with the project.

### How to test your changes?
1. Check out the branch.
2. Create an extension and push it to Shopify.
3. Rename `.env` to `.example.env`
4. Call `shopify extension push --api-key=xxx --api-secret=yyy --registration-id=zz` taking the values from the file renamed in step 3.
5. The push should work.


### Update checklist
